File: src/rentals/src/main/java/e/library/on/containers/rentals/events/Event.java
package e.library.on.containers.rentals.events;

import lombok.Getter;

import java.time.ZonedDateTime;
import java.util.UUID;

@Getter
public abstract class Event {

    protected final UUID id;
    protected final ZonedDateTime createdAt;

    protected Event(UUID id, ZonedDateTime createdAt) {
        this.id = id;
        this.createdAt = createdAt;
    }
}

File: src/rentals/src/main/java/e/library/on/containers/rentals/events/Event.kt
package e.library.on.containers.rentals.events

import java.util.UUID

object Event {
    data class BookExtended(val rentalId: UUID)

    data class BookRented(
        val rentalId: UUID,
        val isbn: String
    )

    data class BookReturned(val bookReturned: UUID)

    data class BookAvailableForRent(val isbn: String)

    data class BookUnavailableForRent(val isbn: String)

    data class BookCopyAdded(
        val isbn: String,
        val bookCopyId: UUID
    )

    data class BookCopyRemoved(val bookCopyId:UUID)
}
